src/applications/Job.java

```diff
@@ -8,6 +8,8 @@
     private int length; // sum of scheduled task times
     private int arrivalTime; // arrival time at current queue
     private int id; // job identifier
+    private int machine; // the machine doing the task
+    private int time; //the time to do the task
 
     // constructor
     Job(int theId) {
```

Comment:
Right now, it seems like jobs can secretly do one of two things.  They have an id, or they might have a machine and a time, but there isn't an obvious way of distinguishing them.  This would make working with them a little risky, since they may or may not have certain fields.